<launch>
  <arg name="input_lanelet_map" default="~/input/vector_map"/>
  <arg name="input_scenario" default="~/input/scenario"/>
  <arg name="output_grid_map" default="~/output/grid_map"/>
  <arg name="output_occupancy_grid" default="~/output/occupancy_grid"/>

  <!-- TODO: change this default -->
  <arg name="client_HAD_map_service" default="/had_maps/HAD_Map_Service"/>

  <node pkg="costmap_generator" exec="costmap_generator" name="costmap_generator" output="screen">
    <remap from="~/input/vector_map" to="$(var input_lanelet_map)" />
    <remap from="~/input/scenario" to="$(var input_scenario)" />
    <remap from="~/output/grid_map" to="$(var output_grid_map)" />
    <remap from="~/output/occupancy_grid" to="$(var output_occupancy_grid)" />
    <remap from="~/client/HAD_Map_Service" to="$(var client_HAD_map_service)" />

    <param name="costmap_frame" value="map" />
    <param name="vehicle_frame" value="base_link" />
    <param name="map_frame" value="map" />

    <param name="update_rate" value="10.0" />

    <param name="use_wayarea" value="true" />

    <param name="grid_min_value" value="0.0" />
    <param name="grid_max_value" value="1.0" />
    <param name="grid_resolution" value="0.2" />
    <param name="grid_length_x" value="70.0" />
    <param name="grid_length_y" value="70.0" />
    <param name="grid_position_x" value="0.0" />
    <param name="grid_position_y" value="0.0" />

    <!--  padding of bounding box for route that determines costmap size  -->
    <param name="route_box_padding_m" value="10.0" />
  </node>
</launch>
